ome_zarr/writer.py

```diff
@@ -13,18 +13,75 @@
 
 LOGGER = logging.getLogger("ome_zarr.writer")
 
+KNOWN_AXES = {"x": "space", "y": "space", "z": "space", "c": "channel", "t": "time"}
 
-def _validate_axes_names(
-    ndim: int, axes: Union[str, List[str]] = None, fmt: Format = CurrentFormat()
-) -> Union[None, List[str]]:
-    """Returns validated list of axes names or raise exception if invalid"""
+
+def _axes_to_dicts(
+    axes: Union[List[str], List[Dict[str, str]]]
+) -> List[Dict[str, str]]:
+    """Returns a list of axis dicts with name and type"""
+    axes_dicts = []
+    for axis in axes:
+        if isinstance(axis, str):
+            axis_dict = {"name": axis}
+            if axis in KNOWN_AXES:
+                axis_dict["type"] = KNOWN_AXES[axis]
```

Comment:
What should happen otherwise? Error? default type?

Reply:
The types are subsequently checked in `_validate_axes_types()`.
Current spec 0.4 PR says `The "axes" MUST contain 2 or 3 entries of "type:space" and MAY contain one additional entry of "type:time" and MAY contain one additional entry of "type:channel" or a null / custom type.`
So a single missing type may not be invalid.

Reply:
but you proposed it should be a requirement in https://github.com/ome/ngff/pull/57/files#r772303584 no ?

Reply:
Ah, yes - good catch! Although that statement is not quite right since we do allow 1 axis of unknown type.
It's probably better to have `type` as optional if we're going to allow it to be `null`.
Again, this PR really needs to have the spec PR to be concrete first.
I'll add a comment there...

Reply:
I updated the statement, and will fix the code here to use  `axis_dict["type"] = None`...

Reply:
Having tried to implement the `axis_dict["type"] = None` option, this breaks `mypy` in a lot of places, and means the `axes` dict is type `List[Dict[str, Optional[str]]]` everywhere, which means that `axes["name"]` can also be `None`. It doesn't make sense for "type" to be required when it can be `None`. Updated comment on PR to reflect that.

Reply:
To summarise, the code here doesn't need to change.